### ISSUE TITLE
Bug: [prefer-as-const] Rule doesn't support class properties

### ISSUE THREAD

**Issue by undsoft** [typescript-eslint/typescript-eslint#5405]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=MYGwhgzhAEAKCWAvRZoG8BQ1oAdIBcwAuaAcjwkNOgF4yKqBuDAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloENktMgHsmyfOii9oI6JHBgAviCVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Pizza {
  pasta: 'pasta' = 'pasta';
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-as-const": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Expect an error and a suggestion to use `as const` instead of explicit type.

### Actual Result

No error.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.31.0` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `X.Y.Z` |

**Comment by undsoft** [comment]
Ready to make a PR.

**Comment by JoshuaKGoldberg** [comment]
![Emperor Palpatine from Star Wars Episode 3 saying "Do it."](https://user-images.githubusercontent.com/3335181/182063415-b137caed-0bee-40a9-b73e-59b8d7229809.gif)
